src/fluree/db/api/query.cljc

```diff
@@ -152,183 +157,110 @@
+      (->> (sort-by flake/t flakes)
+           (partition-by flake/t)
+           (map (fn [t-flakes]
+                  (go-try
+                    (let [assert-flakes (not-empty (filter flake/op t-flakes))
+                          retract-flakes (not-empty (filter (complement flake/op) t-flakes))
+
+                          asserts (<? (json-ld-resp/flakes->res db cache compact fuel 1000000
+                                                                {:wildcard? true, :depth 0} 0 assert-flakes))
+                          retracts (<? (json-ld-resp/flakes->res db cache compact fuel 1000000
+                                                                 {:wildcard? true, :depth 0}
+                                                                 0 retract-flakes))]
+
+                      ;; t is always positive for users
+                      (cond-> {:t (- (flake/t (first t-flakes)))}
+                        asserts (assoc :assert asserts)
+                        retracts (assoc :retract retracts))))))
+           (async/merge)
+           (async/into [])
+           (<?)
+           (sort-by :t)))))
 
 (defn get-history-pattern
   [history]
-  (let [subject (cond (util/subj-ident? history)
-                      [history]
-
-                      (sequential? history)
-                      (if (empty? history)
-                        (throw (ex-info (str "Please specify an subject for which to search history. Provided: " history)
-                                        {:status 400
-                                         :error  :db/invalid-query}))
-                        history)
-
-                      :else
-                      (throw (ex-info (str "History query not properly formatted. Provided: " history)
-                                      {:status 400
-                                       :error  :db/invalid-query})))
-        [s p o t] [(get subject 0) (get subject 1) (get subject 2) (get subject 3)]
-
+  (let [[s p o t]     [(get history 0) (get history 1) (get history 2) (get history 3)]
         [pattern idx] (cond
                         (not (nil? s))
-                        [subject :spot]
+                        [history :spot]
 
                         (and (nil? s) (not (nil? p)) (nil? o))
                         [[p s o t] :psot]
 
                         (and (nil? s) (not (nil? p)) (not (nil? o)))
-                        [[p o s t] :post]
-
-                        :else
-                        (throw (ex-info (str "History query not properly formatted. Must include at least an subject or predicate to query. Provided: " history)
-                                        {:status 400
-                                         :error  :db/invalid-query})))]
+                        [[p o s t] :post])]
     [pattern idx]))
 
+(def History
+  [:map {:registry {::iri [:or :keyword :string]
+                    ::context [:map-of :any :any]}}
+   [:history
+    [:orn
```

Comment:
Should this be `:or`?

Reply:
No, `:orn` is "or with Named branches", so the first element of each `:orn` tuple is the name of that case. It's helpful in parsing. It is a weird name, though!